open_spiel/algorithms/dqn_torch/dqn.h

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPEN_SPIEL_ALGORITHMS_DQN_H_
+#define OPEN_SPIEL_ALGORITHMS_DQN_H_
+
+#include "open_spiel/algorithms/dqn_torch/simple_nets.h"
+
+#include <torch/torch.h>
+
+#include <memory>
+#include <random>
+#include <vector>
+
+#include "open_spiel/abseil-cpp/absl/random/random.h"
+#include "open_spiel/policy.h"
+#include "open_spiel/spiel.h"
+#include "open_spiel/spiel_utils.h"
+#include "open_spiel/utils/circular_buffer.h"
+
+
+namespace open_spiel {
+namespace algorithms {
+namespace torch_dqn {
+
+struct Transition {
+  std::vector<float> info_state;
+  Action action;
+  double reward;
+  std::vector<float> next_info_state;
+  bool is_final_step;
+  std::vector<int> legal_actions_mask;
+};
+
+// DQN Agent implementation in LibTorch.
+class DQN {
+  public: 
+    DQN(bool use_observation,
```

Comment:
A quick comment: use a struct DQNSettings that can be passed to DQN constructor.

Reply:
Thanks for your pointing it out!

Using struct settings would be easier to use!